code/modules/client/client procs.dm

```diff
@@ -806,10 +806,12 @@ var/list/localhost_addresses = list(
 		if(G.can_autofire(O, location, params) && O.is_auto_clickable() && !(G.safety()) && !(G == O))
 			autofire_aiming_at[1] = O
 			autofire_aiming_at[2] = params
+			var/accuracy_dec = 0
 			while(autofire_aiming_at[1])
-				G.Fire(autofire_aiming_at[1], mob, autofire_aiming_at[2], (get_dist(mob, location) <= 1), FALSE)
+				G.Fire(autofire_aiming_at[1], mob, autofire_aiming_at[2], (get_dist(mob, location) <= 1), FALSE, accuracy_dec)
 				mob.set_dir(get_dir(mob, autofire_aiming_at[1]))
 				sleep(G.fire_delay)
+				accuracy_dec += 0.25
```

Comment:
Imho it would be good to limit the accuracy decrease to a maximum value here.
(As it might make things like the mini/machinegun completely useless otherwise)
Ideally that value can be set using a variable on the gun.

Reply:
Wouldn't all you need to do is just stop shooting to regain the accuracy you lost?

Reply:
Limited to 2 max inaccuracy.


> Wouldn't all you need to do is just stop shooting to regain the accuracy you lost?

Might as well use burst then.